packages/web-components/src/avatar/avatar.styles.ts

```diff
@@ -0,0 +1,569 @@
+  colorPaletteSteelBackground2,
+  colorPaletteSteelForeground2,
+  colorPaletteTealBackground2,
+  colorPaletteTealForeground2,
+  fontFamilyBase,
+  fontSizeBase100,
+  fontSizeBase200,
+  fontSizeBase300,
+  fontSizeBase400,
+  fontSizeBase500,
+  fontSizeBase600,
+  fontWeightRegular,
+  fontWeightSemibold,
+  shadow16,
+  shadow28,
+  shadow4,
+  shadow8,
+  strokeWidthThick,
+  strokeWidthThicker,
+  strokeWidthThickest,
+  strokeWidthThin,
+} from '../theme/design-tokens.js';
+
+// TODO: Should all animation constants go to tokens/theme or globals?
+// https://github.com/microsoft/fluentui/issues/16372#issuecomment-778240665
+
+const animationDuration = {
+  duration50: '50ms',
+  duration100: '100ms',
+  duration150: '150ms',
+  duration200: '200ms',
+  duration300: '300ms',
+  duration400: '400ms',
+  duration500: '500ms',
+};
+
+const animationTiming = {
+  ultraFast: animationDuration.duration50,
+  faster: animationDuration.duration100,
+  fast: animationDuration.duration150,
+  normal: animationDuration.duration200,
+  slow: animationDuration.duration300,
+  slower: animationDuration.duration400,
+  ultraSlow: animationDuration.duration500,
+};
+
+const animationLines = {
+  decelerateMax: 'cubic-bezier(0.00,0.00,0.00,1.00)',
+  decelerateMid: 'cubic-bezier(0.10,0.90,0.20,1.00)',
+  decelerateMin: 'cubic-bezier(0.33,0.00,0.10,1.00)',
+  accelerateMax: 'cubic-bezier(1.00,0.00,1.00,1.00)',
+  accelerateMid: 'cubic-bezier(0.90,0.10,1.00,0.20)',
+  accelerateMin: 'cubic-bezier(0.80,0.00,0.78,1.00)',
+  maxEasyEase: 'cubic-bezier(0.80,0.00,0.20,1.00)',
+  easyEase: 'cubic-bezier(0.33,0.00,0.67,1.00)',
+  linear: 'linear',
+};
+
+const animations = {
+  fastOutSlowInMax: animationLines.decelerateMax,
+  fastOutSlowInMid: animationLines.decelerateMid,
+  fastOutSlowInMin: animationLines.decelerateMin,
+  slowOutFastInMax: animationLines.accelerateMax,
+  slowOutFastInMid: animationLines.accelerateMid,
+  slowOutFastInMin: animationLines.accelerateMin,
+  fastEase: animationLines.maxEasyEase,
+  normalEase: animationLines.easyEase,
+  nullEasing: animationLines.linear,
+};
+
+/** Avatar styles
+ * @public
+ */
+export const styles = css`
+  ${display('inline-flex')} :host {
+    position: relative;
+    align-items: center;
+    justify-content: center;
+    flex-shrink: 0;
+    width: 32px;
+    height: 32px;
+    font-family: ${fontFamilyBase};
+    font-weight: ${fontWeightSemibold};
+    font-size: ${fontSizeBase300};
+    border-radius: ${borderRadiusCircular};
+    color: ${colorNeutralForeground3};
+    background-color: ${colorNeutralBackground6};
+  }
+
+  .default-icon,
+  ::slotted(svg) {
+    width: 20px;
+    height: 20px;
+    font-size: 20px;
+  }
+
+  ::slotted(img) {
+    box-sizing: border-box;
+    width: 100%;
+    height: 100%;
+    border-radius: ${borderRadiusCircular};
+  }
+
+  ::slotted(fluent-presence-badge),
+  ::slotted(fluent-badge) {
+    position: absolute;
+    bottom: 0;
+    right: 0;
```

Comment:
What about rtl? Would be cool if we could leverage `inset-inline` here.

Reply:
I think it depends on what version of Safari we need to pin on.

Reply:
I'd suspect Safari 17 will be released sometime this year (There seems to be a major version released every year since 2013). Support for `inset-inline` goes back to Safari 14.1.

Reply:
Yep, but I think we need a better handle on moving the matrix before just making the change. Our current policy in FAST for instance is "last 4 major except Safari which starts at 13.1". So it's already an exception :)